js/runner.go

```diff
@@ -388,20 +388,25 @@ func (r *Runner) HandleSummary(ctx context.Context, summary *lib.Summary) (map[s
 		return nil, err
 	}
 
-	handleSummaryFn := goja.Undefined()
-	fn := vu.getExported(consts.HandleSummaryFn)
-	if _, ok := goja.AssertFunction(fn); ok {
-		handleSummaryFn = fn
-	} else if fn != nil {
-		return nil, fmt.Errorf("exported identifier %s must be a function", consts.HandleSummaryFn)
-	}
-
 	go func() {
 		<-summaryCtx.Done()
 		vu.Runtime.Interrupt(context.Canceled)
 	}()
 	vu.moduleVUImpl.ctx = summaryCtx
 
+	callbackResult := goja.Undefined()
+	fn := vu.getExported(consts.HandleSummaryFn)
+	if fn != nil {
+		if handleSummaryFn, ok := goja.AssertFunction(fn); ok {
+			callbackResult, _, _, err = vu.runFn(summaryCtx, false, handleSummaryFn, nil, vu.Runtime.ToValue(summaryDataForJS))
+			if err != nil {
+				errext.Fprint(r.preInitState.Logger, err)
+			}
+		} else if fn != nil {
+			return nil, fmt.Errorf("exported identifier %s must be a function", consts.HandleSummaryFn)
+		}
```

Comment:
nit: we usually prefer to nest less. In this case you can check that `!ok` first and error out with this error.  

If we are in this if `fn != nil` is already true ;)

Reply:
I also prefer it! It's just that as the code was there more less like that, I simply forgot to simplify it, good catch, thanks!